src/scenegraph/index.ts

```diff
@@ -257,63 +257,81 @@ function parseChildren(element: XmlElement, children: ComponentNode[]): void {
 
 async function getScripts(
     node: XmlDocument,
-    xmlPath: string,
+    nodeDef: ComponentDefinition,
     rootDir: string
 ): Promise<ComponentScript[]> {
     let scripts = node.childrenNamed("script");
     let componentScripts: ComponentScript[] = [];
 
     for (let script of scripts) {
-        let absoluteUri: URL;
-        let posixRoot = rootDir.replace(/[\/\\]+/g, path.posix.sep);
-        let posixPath = xmlPath.replace(/[\/\\]+/g, path.posix.sep);
-
-        try {
-            if (process.platform === "win32") {
-                posixRoot = posixRoot.replace(/^[a-zA-Z]:/, "");
-                posixPath = posixPath.replace(/^[a-zA-Z]:/, "");
-            }
-            absoluteUri = new URL(
-                script.attr.uri,
-                `pkg:/${path.posix.relative(posixRoot, posixPath)}`
-            );
-        } catch (err) {
-            let file = await readFile(xmlPath, "utf-8");
-
-            let tag = file.substring(script.startTagPosition, script.position);
-            let tagLines = tag.split("\n");
-            let leadingLines = file.substring(0, script.startTagPosition).split("\n");
-            let start = {
-                line: leadingLines.length,
-                column: columnsInLastLine(leadingLines),
-            };
-
+        if (script.attr.uri && script.val) {
             return Promise.reject({
                 message: BrsError.format(
-                    `Invalid path '${script.attr.uri}' found in <script/> tag`,
-                    {
-                        file: xmlPath,
-                        start: start,
-                        end: {
-                            line: start.line + tagLines.length - 1,
-                            column: start.column + columnsInLastLine(tagLines),
-                        },
-                    }
+                    `<script> element cannot contain both internal and external source`,
```

Comment:
What's the reasoning for this error? Also, it's not clear to me what _internal_ or _external_ source even means. Could the error be more helpful? Also, perhaps a comment around this line explaining what the problem actually is (or why the limitation exists)?

Reply:
<img width="1291" alt="image" src="https://github.com/user-attachments/assets/c11e26be-4ede-4369-a495-ae8b2b603974" />

This exception was copy pasted from Roku telnet debug, if you try to define two scripts one internal to the XML and other external.

Reply:
Okay. Makes sense. I think it would be useful to add some of this info into the code as a comment for future readers.